mmdet/core/results/results.py

```diff
@@ -0,0 +1,273 @@
+               list(self._meta_info_field.values())
+
+    def items(self):
+        items = []
+        for k, v in self._results_field.items():
+            items.append((k, v))
+        for k, v in self._meta_info_field.items():
+            items.append((k, v))
+        return items
+
+    def __setattr__(self, name, val):
+        if name.startswith('_'):
+            super().__setattr__(name, val)
+        elif name in self._meta_info_field:
+            raise AttributeError(f'{name} is used in meta information,'
+                                 f'which is unmodifiable')
+        else:
+            self.set(name, val)
+
+    __setitem__ = __setattr__
+
+    def __getattr__(self, name):
+        if name in self._meta_info_field:
+            return self._meta_info_field[name]
+        elif name in self._results_field:
+            return self._results_field[name]
+        else:
+            raise AttributeError(f'Can not find {name}')
+
+    __getitem__ = __getattr__
+
+    def set(self, name, value):
+        self._results_field[name] = value
+
+    def get(self, name):
+        return self._results_field[name]
+
+    def has(self, name):
+        return name in self._results_field or name in self._meta_info_field
+
+    def remove(self, name):
+        del self._results_field[name]
+
+    def results(self):
+        return copy.deepcopy(self._results_field)
+
+    def meta_info(self):
+        return copy.deepcopy(self._meta_info_field)
+
+    # Tensor-like methods
+    def to(self, *args, **kwargs):
+        new_instance = copy.deepcopy(self)
+        for k, v in new_instance.items():
+            if isinstance(v, torch.Tensor):
+                new_instance[k] = v.to(*args, **kwargs)
+        return new_instance
+
+    # Tensor-like methods
+    def cpu(self):
+        new_instance = self.__deepcopy__()
+        for k, v in new_instance.items():
+            if isinstance(v, torch.Tensor):
+                new_instance[k] = v.cpu()
+        return new_instance
+
+    # Tensor-like methods
+    def cuda(self):
+        new_instance = self.__deepcopy__()
+        for k, v in self.__dict__.items():
+            if isinstance(v, dict):
+                new_instance[k] = copy.deepcopy(v)
+        for k, v in new_instance.items():
+            if isinstance(v, torch.Tensor):
+                new_instance[k] = v.cuda()
+        return new_instance
+
+    # Tensor-like methods
+    def detach(self):
+        new_instance = self.__deepcopy__()
+        for k, v in new_instance.items():
+            if isinstance(v, torch.Tensor):
+                v.requires_grad_(False)
+        return new_instance
+
+    # Tensor-like methods
+    def numpy(self):
+        new_instance = self.__deepcopy__()
+        for k, v in new_instance.items():
+            if isinstance(v, torch.Tensor):
+                new_instance[k] = v.numpy()
+        return new_instance
+
+    def __str__(self):
+        repr = f'{self.__class__.__name__}: \n'
```

Comment:
Suggest using the nicerepr implemented in mmdet [here](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/utils/util_mixins.py)